bitbots_vision/src/bitbots_vision/vision_modules/color.py

```diff
@@ -284,22 +275,92 @@ def mask_image(self, image):
         # type: (np.array) -> np.array
         """
         Creates a color mask (0 for not in color range and 255 for in color range)
-        and publishes 'mask_img' and 'mask_img_dyn'-messages.
+        and publishes 'field_mask' Image-messages.
+
+        :param np.array image: image to mask
+        :return np.array: masked image
+        """
+        mask = VisionExtensions.maskImg(image, self.color_space)
+
+        # toggle publishing of 'field_mask'-messages   
+        if self.config['vision_field_mask_image']:
+            self.imagepublisher.publish(self.bridge.cv2_to_imgmsg(mask, '8UC1'))
+
+        return mask
+
+
+class DynamicPixelListColorDetector(PixelListColorDetector):
+    """
+    DynamicPixelListColorDetector is a ColorDetector, that is based on a lookup table of color values.
+    The color space is initially loaded from color-space-file at color_path (in config)
+    and optionally adjustable to changing color conditions (dynamic color space).
+    The color space is represented by boolean-values for RGB-color-values.
+
+    Subscribes to: 'color_space'-message
+    Publishes: 'field_mask' and 'dynamic_field_mask'-messages
+    """
+
+    def __init__(self, debug_printer, package_path, config, primary_detector=False):
+        # type:(DebugPrinter, str, dict, bool) -> None
+        """
+        Initiating of DynamicPixelListColorDetector.
+
+        :param DebugPrinter debug_printer: debug-printer
+        :param str package_path: path of package
+        :param dict config: vision config
+        :param bool primary_detector: true if is primary color detector
+            (only detector held by vision should be True) (Default: False)
+        :return: None
+        """
+        super(DynamicPixelListColorDetector, self).__init__(debug_printer, package_path, config)
+
+        self.primary_detector = primary_detector
+
+        self.base_color_space = np.copy(self.color_space)
+
+        # toggle publishing of mask_img msg
+        self.publish_field_mask_img_msg = self.config['vision_field_mask_image']
+        
+        # toggle publishing of mask_img_dyn msg with dynamic color space
+        self.publish_dyn_field_mask_msg = self.config['dynamic_color_space_field_mask_image']
+
+        # toggle use of dynamic color space
+        self.dynamic_color_space_turned_on = self.config['dynamic_color_space']
+
+        # Subscribe to 'color_space'-messages from DynamicColorSpace
+        self.color_space_subscriber = rospy.Subscriber(
+            'color_space',
+            ColorSpace,
+            self.color_space_callback,
+            queue_size=1,
+            buff_size=2**20)
+
+        # Register publisher for 'mask_image_dyn'-messages
+        self.imagepublisher_dyn = rospy.Publisher(
+            "/dynamic_field_mask",
```

Comment:
This should be configurable.

Reply:
closed with 1d1bfc5babf81c18a5b762e4508d002de9d94d5c and c7c947b904fc75ac6834c9e8ae846a67bbeb3df5